src/modules/Core.class.js

```diff
@@ -0,0 +1,319 @@
+'use strict';
+import {
+  GAME_STATUSES,
+  MESSAGES_KEYS,
+  FIELD_ROW_LENGTH,
+} from '../lib/constants';
+
+import {
+  FIELD_HTML_TEMPLATE,
+  HEADER_HTML_TEMPLATE,
+  MESSAGE_LOSE_HTML_TEMPLATE,
+} from '../lib/templates';
+
+import {
+  generateCellValue,
+  generateRandomPosition,
+  horizontalShift,
+  rotateFieldForward,
+  verticalShift,
+} from '../lib/helpers';
+
+class Core {
+  field = null;
+  score = 0;
+  fieldControls = {};
+
+  status = null;
+  messages = {};
+
+  #triggerMessage({ key, show }) {
+    if (show) {
+      this.messages[key].classList.remove('hidden');
+    } else {
+      this.messages[key].classList.add('hidden');
+    }
+  }
+
+  #updateField(newField) {
+    this.field = newField;
+  }
+
+  #updateScore(score) {
+    this.score += score;
+  }
+
+  #createNewField() {
+    const newField = Array.from({ length: FIELD_ROW_LENGTH }, () => [
+      0, 0, 0, 0,
+    ]);
+
+    this.#updateField(newField);
+  }
+
+  #seed() {
+    const field = this._getField();
+
+    const [firstCellPosX, firstCellPosY] = generateRandomPosition({ field });
+    const [secondCellPosX, secondCellPosY] = generateRandomPosition({
+      field,
+      exclude: [firstCellPosX, firstCellPosY],
+    });
+
+    field[firstCellPosY][firstCellPosX] = generateCellValue();
+    field[secondCellPosY][secondCellPosX] = generateCellValue();
+
+    this.#updateField(field);
+  }
+
+  #initialRender(rootRef) {
+    rootRef.classList.add('container');
+
+    const headerNode = document.createElement('div');
```

Comment:
Your structure is nice, but why just don't you use the initial HTML layout (in index.html) instead of setting it via JS?

Reply:
I like this JS-built structure because all you need is just in the box. Just call new Game() class and enjoy, without any additional HTML.